### PR TITLE
View-Live-DropDown

### PR DESCRIPTION
**Pull request by partpa**
#### What's this PR do? Any additional context?
 -  It creates dropdown for each course. Dropdown will have all site's display name for which course is 
    targeted to. 

#### Where should the reviewer start?
- Open CMS. View Live will redirect to course organization.

#### How can this be manually tested? (brief repro steps)
-Before : View Live redirects based on organization of course.
-After   : View Live button will redirect based on selected value of dropdown. 

#### What are the relevant TFS items? (list id numbers)
Feature : 99205

#### Definition of done:
- [ ] Title of the pull request is clear and informative
- [ ] Add pull request hyperlink to relevant TFS items
- [ ] For large or complex change: schedule an in-person review session

[//]: # ( todo: Is there appropriate test coverage? )
[//]: # ( todo: Does this PR require a new Selenium test? )
[//]: # ( todo: Is there appropriate logging/monitoring included? )

#### Reminders BEFORE merging
1. Get at least two approvals
1. If you're merging into the development branch then "flatten" or "squash" commits
1. If merging from development into master then don't "flatten" or "squash" commits

#### Reminders AFTER merging
1. Delete the remote branch
1. Resolve relevant TFS items
1. (reverse merge) If you merged into master then check to see if there are any changes in master that can be merged down to the development branch (like hotfixes, etc)

[//]: # ( todo: If you merged into development branch then verify change in our "rolling deployment" environment. Then notify stakeholders interested in or involved with the change )

[//]: # ( fyi: This content was heavily inspired by )
[//]: # ( 1 Our team's policies and processes )
[//]: # ( 3 https://github.com/sprintly/sprint.ly-culture/blob/master/pr-template.md )
[//]: # ( 4 The book "The Checklist Manifesto: How to Get Things Right" by Atul Gawande )
[//]: # ( 5 https://github.com/Azure/azure-event-hubs/blob/master/.github/PULL_REQUEST_TEMPLATE.md )

### CODE REVIEW

**Comment by partpa** [comment]
I will add Test function soon.

File: cms/static/js/base.js [hunk 1]
```diff
@@ -98,6 +98,9 @@ require([
                 });
                 dropdownMenuView.postRender();
             }
+            // Change link based on selected site
+            $('.view-button').bind('click', selectSite);
```
Comment (sdolenc) [inline cms/static/js/base.js]: use on instead of bind

File: cms/static/js/base.js [hunk 2]
```diff
@@ -133,4 +136,12 @@ require([
             (e).preventDefault();
             $(this).closest('.wrapper-alert').removeClass('is-shown');
         }
+        function selectSite() {
+            var selectedSite = $(this).parent().parent().find('.site_name');
```
Comment (sdolenc) [inline cms/static/js/base.js]: parent().parent() feel brittle. do we know the id, class, or jquery handle of another DOM element that contains the .site_name element you're looking for?

File: cms/static/js/base.js [hunk 3]
```diff
@@ -133,4 +136,12 @@ require([
...
+            var site_name = selectedSite.find("option:selected").val();
```
Comment (btelnes) [inline cms/static/js/base.js]: Nit: Use camel casing.  Change site_name to siteName to be consistent with the rest of the file.

File: cms/static/js/base.js [hunk 4]
```diff
@@ -133,4 +136,12 @@ require([
...
+            var site = $(this).attr('href');
+            var href = site.replace(site.split('/')[2], site_name);
```
Comment (sdolenc) [inline cms/static/js/base.js]: URI building is a hard problem to get right. url string manipulation should be handled by JS, a library, or wrapped in functions that we own. if we go for that last option then we need unit tests for it

**Comment by sdolenc** [comment]
i saw that you want to add tests to this (via a comment earlier). that's awesome! I think this is a great candidate for testing. I recommend you peer develop those with Kabir, Brian, me, or someone else since that's something we all need to get better at as a team

**Pull request merged**